test/server/TypeDBClusterRunner.java

```diff
@@ -18,34 +18,70 @@
 
 package com.vaticle.typedb.common.test.server;
 
-import com.vaticle.typedb.common.collection.Triple;
-
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
 import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.TimeoutException;
 
-import static com.vaticle.typedb.common.collection.Collections.list;
-import static com.vaticle.typedb.common.collection.Collections.triple;
+import static com.vaticle.typedb.common.collection.Collections.map;
+import static com.vaticle.typedb.common.collection.Collections.set;
 
 public class TypeDBClusterRunner extends TypeDBRunner {
 
-    private final Triple<Integer, Integer, Integer> ports;
-    private final List<Triple<Integer, Integer, Integer>> peerPorts;
+    public static final String OPT_ADDR = "--server.address";
+    private static final String OPT_INTERNAL_ADDR_ZMQ = "--server.internal-address.zeromq";
+    private static final String OPT_INTERNAL_ADDR_GRPC = "--server.internal-address.grpc";
+    public static final String OPT_PEERS_ADDR = "--server.peers.server-{peer-id}.address";
+    private static final String OPT_PEERS_INTERNAL_ADDR_ZMQ = "--server.peers.server-{peer-id}.internal-address.zeromq";
+    private static final String OPT_PEERS_INTERNAL_ADDR_GRPC = "--server.peers.server-{peer-id}.internal-address.grpc";
 
-    public TypeDBClusterRunner(Triple<Integer, Integer, Integer> ports, List<Triple<Integer, Integer, Integer>> peerPorts) throws InterruptedException, TimeoutException, IOException {
-        super();
-        this.ports = ports;
-        this.peerPorts = peerPorts;
+    private final Ports ports;
+    private final Set<Ports> peers;
+    private final Map<String, String> remainingServerOpts;
+
+    public static TypeDBClusterRunner create() throws InterruptedException, TimeoutException, IOException {
+        int port = ThreadLocalRandom.current().nextInt(40000, 60000);
+        Ports server = new Ports(port, port+1, port+2);
+        return TypeDBClusterRunner.create(server);
```

Comment:
you shouldn't need to call `TypeDBClusterRunner.create(server)`, just call `create(server)` right?